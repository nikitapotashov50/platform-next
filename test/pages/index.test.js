import test from 'ava'
import render from 'react-test-renderer'
import React from 'react'
import { mount } from 'enzyme'
import IndexPage from '../../pages/index'

test('Index page snapshot', t => {
  const tree = render.create(<IndexPage />).toJSON()
  t.snapshot(tree)
})

test('Index page state', t => {
  const wrapper = mount(<IndexPage />)
  t.deepEqual(wrapper.state(), {})

  const changeEvent = {
    target: {
      value: 'test'
    }
  }

  wrapper.find('input').simulate('change', changeEvent)
  t.deepEqual(wrapper.state(), {
    login: 'test'
  })
})
