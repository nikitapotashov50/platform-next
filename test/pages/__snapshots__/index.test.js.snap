// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page 1`] = `
<div>
  <h1>
    Главная
  </h1>
  <input
    onChange={[Function]}
    placeholder="Логин"
    type="text"
  />
  <button
    onClick={[Function]}
  >
    ВОЙТИ
  </button>
</div>
`;

exports[`Index page snapshot 1`] = `
<div>
  <h1>
    Главная
  </h1>
  <input
    onChange={[Function]}
    placeholder="Логин"
    type="text"
  />
  <button
    onClick={[Function]}
  >
    ВОЙТИ
  </button>
</div>
`;
